### PR TITLE
CODE-23 add possible errors, best practices, stylistic issues eslint …

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
## Description
In an effort towards common standards across the digital services teams we are trying to consolidate some of the best practices from each team. This PR adds a lot new rules that eslint provides.

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests
No tests were included as this is a just linting configuration for which we don't have tests.

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by NagaPradeepGajula-osi** [comment]
@edgibbs First thanks for pointing out towards the cwds/development-practices repo didn't know that we had one. We don't have any reference of these linting rules in there.
I will get back to my team about this and try to get it updated maybe.

And as per the sync with other JS projects, initially we want to get through CANS and rollout it to other projects one after the other ourselves as the teams have their own priorities. We are aiming to get IDM very soon.

**Comment by edgibbs** [comment]
OK, approving since this is the alpha project to try this out.

**Comment by NagaPradeepGajula-osi** [comment]
yes I did. These rules have no errors with the existing cans code. There are going to be more in future that are mentioned in the story.

File: .eslintrc.js [hunk 1]
```diff
@@ -32,14 +45,73 @@ module.exports = {
...
-    'react/no-unused-state': 2,
+    'key-spacing': [2, {'beforeColon': false, 'afterColon': true}],
+    'linebreak-style': [2, 'unix'],
+    'new-cap': [2, { 'capIsNew': false }],
+    'new-parens': [2],
+    'no-alert': [2],
+    'no-array-constructor': [2],
+    'no-caller': [2],
+    'no-class-assign': [2],
+    'no-console': 0,
```
Comment (wmitchell-cambria) [inline .eslintrc.js]: Double checking that this is intended to be off.

Comment (NagaPradeepGajula-osi) [reply .eslintrc.js]: Ah! good catch this should be enabled. The idea was to have the lint call out the console logs before opening a PR and reviews requesting it to remove them. Will be updating the rule.

**Comment by NagaPradeepGajula-osi** [comment]
Thanks for bringing this up @wmitchell-cambria 
1. I have already mentioned about rolling out these rules above and keeping in them in sync across projects. Also, for the second part of your 1st concern, we can define our own ruleset and thats how we have our intake configured and I am trying to replicate the same here.
2. Realized that CANS already has standard eslint config as extends which already takes care about most of the rules I have here. So, I either should get rid of the standard extend and keep the rules open to be able config to project standards liking or Remove the custom rules and have the standard extend. I would love to have input on this form @wmitchell-cambria @ratneshraval @edgibbs.
3. The un-necessary rules or the one's most needed are not included in this PR and were documented in a excel and is up for discussion with leads to decide.

**Comment by NagaPradeepGajula-osi** [comment]
Hi @denys-davydov I wanted to keep this PR open as I was waiting for a response from the leads, which you can notice in my previous comment point 2. Since all checks and approvals were there and it was ready to merge there is no harm in merging it, but as original owner of it I have more context to not merge it. I am also guilty on not leaving a WIP or Don't Merge label. So, please let's not do it again in future.

**Comment by denys-davydov** [comment]
Hi @NagaPradeepGajula-osi 
Since there were 3 developers in CANS we've been merging after 2 approvals, the second approver merged it like "automatically". As an author, you can mark a PR with WIP in the title so it is not merged. Yesterday we discussed it within Team Bee, now we have 3 teams and even some developers from other teams are working on the same code base so our old practice seemed to be not applicable anymore. We will wait for the PR author to merge. Sorry for merging it, however.

**Pull request merged**